analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentTask.java

```diff
@@ -644,4 +656,124 @@ private void writeRepositorySettingsFile(Map<ArtifactRepository, Path> repositor
+                .entrySet();
+
+        if (configuration.versionModificationEnabled()) {
+            for (Map.Entry<Project, Map<RelaxedProjectVersionRef, ProjectVersionRef>> entry : entrySet) {
+                final Project name = entry.getKey();
+                final Map<RelaxedProjectVersionRef, ProjectVersionRef> allModuleDependencies = entry.getValue();
+                final ManipulationModel correspondingModule = alignmentModel.findCorrespondingChild(name);
+                final String group = correspondingModule.getGroup().isEmpty() ? alignmentModel.getGroup()
+                        : correspondingModule.getGroup();
+                final String artifact = correspondingModule.getName();
+                final String ga = group + ":" + artifact;
+                final String originalModuleGav = ga + ":" + (correspondingModule.getOriginalVersion() == null
+                        ? ""
+                        : correspondingModule.getOriginalVersion());
+                final String newModuleGav = ga + ":" + correspondingModule.getVersion();
+                append(builder, "\tDependencies : {} --> {}", originalModuleGav, newModuleGav);
+                final ModulesItem module = new ModulesItem();
+                module.getGav().setOriginalGAV(originalModuleGav);
+                module.getGav().setPVR(SimpleProjectVersionRef.parse(newModuleGav));
+                modules.add(module);
+                final Map<String, ProjectVersionRef> dependencies = module.getDependencies();
+                final boolean reportNonAligned = configuration.reportNonAligned();
+                allModuleDependencies.forEach((d, p) -> {
+                    final ProjectVersionRef newDependencyVersion = correspondingModule.getAlignedDependencies()
+                            .get(d.toString());
+                    logger.debug("In module {} with GAV {} found a replacement version of {}",
+                            correspondingModule.getProjectPathName(), p, newDependencyVersion);
+
+                    if (newDependencyVersion == null) {
+                        if (reportNonAligned) {
+                            nonAligned.add(d);
+                        }
+                    } else {
+                        final String originalDependencyGav = String.format("%s:%s:%s", p.getGroupId(), p.getArtifactId(),
+                                p.getVersionString());
+                        final String newDependencyGav = String.format("%s:%s:%s", newDependencyVersion.getGroupId(),
+                                newDependencyVersion.getArtifactId(), newDependencyVersion.getVersionString());
+                        append(builder, "\tDependencies : {} --> {}", originalDependencyGav, newDependencyGav);
+                        dependencies.put(d.toString(), newDependencyVersion);
+                    }
+                });
+            }
```

Comment:
Couple of questions:

1. Is there a reason sub-modules are treated as dependencies? Shouldn't they be separated out as its rather confusing to see a submodule be a dependency?
2. Do you not (currently?) have a capability to separate out dependencies from different sub-modules?

Reply:
So, it's not really intentional if I am misunderstanding something here where PME is also separating it out. 

I tried to follow what I saw in PME, with this loop being the equivalent of PMEs `compareDependencies( DEPENDENCIES, ...)`, or so I thought.

Because the loop does look up relative to `correspondingModule`, I thought it is separated by module.

Reply:
I think we're getting confused on terminology. I see in your report, a sub-module appearing as a 'dependency'?

Reply:
OK, it's now printing `"------------------- project"` instead. To me, the text now looks good.

Reply:
Yeah looks good!